java/sage/MiniClientSageRenderer.java

```diff
@@ -1328,6 +1328,33 @@ private boolean checkForAudioOutputChangeUpdates()
     return true;
   }
 
+  private boolean checkForMenuHintUpdates()
+  {
+    String hintText;
+    synchronized (menuHint) {
+      hintText=menuHint.format();
+      menuHint.clear();
+    }
+    if (hintText!=null)
```

Comment:
That's a good point, but, calling hasHint outside of the syncronized block seems to defeat the purpose of actually synchronizing in the first place.  I think that format() should return null if there is no hint, and that way the entire acess to menuHint remains within a sync block.  Otherwise,  I have to put al the code if if(menuHint.hasHint()) inside of a sync block and my feeling was that would be holding the lock for too long.